tests/v2/validation/fleet/public_gitrepo_test.go

```diff
@@ -123,39 +116,40 @@ func (f *FleetPublicRepoTestSuite) TestGitRepoDeployment() {
 }
 
 func (f *FleetPublicRepoTestSuite) TestDynamicGitRepoDeployment() {
-	defer f.session.Cleanup()
+
+	testSession := session.NewSession()
+	defer testSession.Cleanup()
+	client, err := f.client.WithSession(testSession)
+	require.NoError(f.T(), err)
 
 	dynamicGitRepo := fleet.GitRepoConfig()
 	require.NotNil(f.T(), dynamicGitRepo)
 
 	if len(dynamicGitRepo.Spec.Targets) < 1 {
 		dynamicGitRepo.Spec.Targets = []v1alpha1.GitTarget{
 			{
-				ClusterName: f.client.RancherConfig.ClusterName,
+				ClusterName: client.RancherConfig.ClusterName,
 			},
 		}
 	}
 
-	fleetVersion, err := fleet.GetDeploymentVersion(f.client, fleet.FleetControllerName, fleet.LocalName)
-	require.NoError(f.T(), err)
-
-	chartVersion, err := f.client.Catalog.GetLatestChartVersion(fleet.FleetName, catalog.RancherChartRepo)
+	fleetVersion, err := fleet.GetDeploymentVersion(client, fleet.FleetControllerName, fleet.LocalName)
 	require.NoError(f.T(), err)
-	require.Contains(f.T(), chartVersion, fleetVersion[1:])
 
-	f.client, err = f.client.ReLogin()
-	require.NoError(f.T(), err)
-
-	logrus.Info("Deploying dynamic gitRepo: ", dynamicGitRepo.ObjectMeta)
-	logrus.Info("Deploying dynamic gitRepo: ", dynamicGitRepo.Spec)
+	f.Run("fleet "+fleetVersion, func() {
+		client, err = client.ReLogin()
+		require.NoError(f.T(), err)
 
-	gitRepoObject, err := extensionsfleet.CreateFleetGitRepo(f.client, dynamicGitRepo)
-	require.NoError(f.T(), err)
+		logrus.Info("Deploying dynamic gitRepo: ", dynamicGitRepo.ObjectMeta)
+		logrus.Info("Deploying dynamic gitRepo: ", dynamicGitRepo.Spec)
```

Comment:
Noticed we have same log comments but different parameters. is this intentional. if yes, can we combine both in one statement?